source/blog/2017-07-18-running-kubernetes-on-fedora-atomic-26.html.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Running Kubernetes on Fedora Atomic 26
+author: jbrooks
+date: 2017-07-18 23:12:57 UTC
+tags: kubernetes, ansible, kubeadm
+comments: true
+published: true
+---
+
+Fedora Atomic 26 relies on Kubernetes for automating deployment, scaling, and operations of application containers across clusters of hosts. Below is an overview of your options for installing and configuring Kubernetes clusters on Fedora Atomic hosts.
+
+If you're looking to get up and running as quickly as possible with a Fedora Atomic-hosted Kubernetes cluster, skip ahead to the "Kubeadm Deployment" section below.
+
+# Install Kubernetes
+
+## Use Built-in Packages
+
+Fedora Atomic 26 ships with Kubernetes packages baked into the system image. The specific version of Kubernetes included matches the latest release marked stable for f26 in Fedora's [updates system](https://bodhi.fedoraproject.org/updates/?packages=kubernetes&release=F26). If this is the version you wish to run, you can move on to the Manual Deployment, Ansible Deployment or Kubeadm Deployment sections.
+
+### Updates and Testing Packages
+
+If there is a newer stable Kubernetes version available that hasn't yet appeared in a two-weekly Fedora Atomic release, you can access it by rebasing to the updates ref of Fedora Atomic, which is recomposed each night to track the latest stable packages:
+
+```bash
+# rpm-ostree rebase fedora-atomic:fedora/26/x86_64/updates/atomic-host -r
+```
+
+Similarly, if there is a newer Kubernetes version available in Fedora's updates-testing repository, you can access it by rebasing to the testing ref of Fedora Atomic, which is recomposed each night to track the latest testing packages:
+
+```bash
+# rpm-ostree rebase fedora-atomic:fedora/26/x86_64/testing/atomic-host -r
+```
+
+## Use System Containers
+
+You can also install and Kubernetes using [system containers](http://www.projectatomic.io/blog/2016/09/intro-to-system-containers/), a new approach that will eventually replace baked-in Kubernetes packages in the atomic host. You can begin trying it out now, however:
+
+### Run on your kubernetes master
+
+```bash
+# atomic install --system --system-package=no --name kube-apiserver registry.fedoraproject.org/f26/kubernetes-apiserver
+
+# atomic install --system --system-package=no --name kube-controller-manager registry.fedoraproject.org/f26/kubernetes-controller-manager
+
+# atomic install --system --system-package=no --name kube-scheduler registry.fedoraproject.org/f26/kubernetes-scheduler
+```
+
+### Run on your kubernetes node(s)
+
+```bash
+# atomic install --system --system-package=no --name kubelet registry.fedoraproject.org/f26/kubernetes-kubelet
+
+# atomic install --system --system-package=no --name kube-proxy registry.fedoraproject.org/f26/kubernetes-proxy
+```
```

Comment:
should we add a *note* somewhere in here mentioning where system containers are stored on the filesystem?